### PR TITLE
upgrade Azure disk csi driver to v1.25.0

### PR DESCRIPTION
**Pull request by andyzhangx**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
upgrade Azure disk csi driver to v1.25.0

**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:
- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
upgrade Azure disk csi driver to v1.25.0
```

### CODE REVIEW

File: vhdbuilder/packer/generate-windows-vhd-configuration.ps1 [hunk 1]
```diff
@@ -48,8 +48,8 @@ if ($windowsSku -eq "2019") {
             "mcr.microsoft.com/oss/kubernetes-csi/livenessprobe:v2.6.0",
             "mcr.microsoft.com/oss/kubernetes-csi/csi-node-driver-registrar:v2.4.0",
             "mcr.microsoft.com/oss/kubernetes-csi/csi-node-driver-registrar:v2.5.0",
-            "mcr.microsoft.com/oss/kubernetes-csi/azuredisk-csi:v1.23.0",
```
Comment (AbelHu) [inline vhdbuilder/packer/generate-windows-vhd-configuration.ps1:51]: Has we upgraded it from 1.23 to 1.24 in production? If no, we should still keep the old version until v1.23.0 is not used in production.

Comment (andyzhangx) [reply vhdbuilder/packer/generate-windows-vhd-configuration.ps1:51]: yes, 1.24 is current version in production now which is released one month ago.

Comment (AbelHu) [reply vhdbuilder/packer/generate-windows-vhd-configuration.ps1:51]: I think that we should cache the new image version for better perf next time and then update the new version in production.

**Review by AbelHu — APPROVED** [review]
Has we upgraded it from 1.23 to 1.24 in production? If no, we should still keep the old version until v1.23.0 is not used in production.

**Pull request merged**